neworigins/extra.cpp

```diff
@@ -0,0 +1,1481 @@
+// of the License, or (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with this program, in the file license.txt. If not, write
+// to the Free Software Foundation, Inc., 59 Temple Place - Suite 330,
+// Boston, MA 02111-1307, USA.
+//
+// See the Atlantis Project web page for details:
+// http://www.prankster.com/project
+//
+// END A3HEADER
+//
+// This file contains extra game-specific functions
+//
+#include "game.h"
+#include "gamedata.h"
+#include "quests.h"
+#include <string>
+#include <iterator>
+
+#define RELICS_REQUIRED_FOR_VICTORY	7
+#define MINIMUM_ACTIVE_QUESTS		3
+#define MAXIMUM_ACTIVE_QUESTS		10
+#define QUEST_EXPLORATION_PERCENT	100
+#define QUEST_SPAWN_RATE		4
+#define QUEST_SPAWN_CHANCE		40
+#define MAX_DESTINATIONS		5
+
+int Game::SetupFaction( Faction *pFac )
+{
+	pFac->unclaimed = Globals->START_MONEY + TurnNumber() * 100;
+
+	if (pFac->noStartLeader) {
+		return 1;
+	}
+
+	//
+	// Set up first unit.
+	//
+	Unit *temp2 = GetNewUnit( pFac );
+	temp2->SetMen(I_LEADERS, 1);
+	pFac->DiscoverItem(I_LEADERS, 0, 1);
+	temp2->reveal = REVEAL_FACTION;
+
+	//
+	// Set up magic
+	//
+	temp2->type = U_MAGE;
+	temp2->Study(S_FORCE, 30);
+	temp2->Study(S_PATTERN, 30);
+	temp2->Study(S_SPIRIT, 30);
+	temp2->Study(S_GATE_LORE, 30);
+	
+
+	if (Globals->UPKEEP_MINIMUM_FOOD > 0)
+	{
+		if (!(ItemDefs[I_FOOD].flags & ItemType::DISABLED)) {
+			temp2->items.SetNum(I_FOOD, 6);
+			pFac->DiscoverItem(I_FOOD, 0, 1);
+		} else if (!(ItemDefs[I_FISH].flags & ItemType::DISABLED)) {
+			temp2->items.SetNum(I_FISH, 6);
+			pFac->DiscoverItem(I_FISH, 0, 1);
+		} else if (!(ItemDefs[I_LIVESTOCK].flags & ItemType::DISABLED)) {
+			temp2->items.SetNum(I_LIVESTOCK, 6);
+			pFac->DiscoverItem(I_LIVESTOCK, 0, 1);
+		} else if (!(ItemDefs[I_GRAIN].flags & ItemType::DISABLED)) {
+			temp2->items.SetNum(I_GRAIN, 2);
+			pFac->DiscoverItem(I_GRAIN, 0, 1);
+		}
+		temp2->items.SetNum(I_SILVER, 10);
+	}
+
+	ARegion *reg = NULL;
+	if (pFac->pStartLoc) {
+		reg = pFac->pStartLoc;
+	} else if (!Globals->MULTI_HEX_NEXUS) {
+		reg = (ARegion *)(regions.First());
+	} else {
+		ARegionArray *pArr = regions.GetRegionArray(ARegionArray::LEVEL_NEXUS);
+		while(!reg) {
+			reg = pArr->GetRegion(getrandom(pArr->x), getrandom(pArr->y));
+		}
+	}
+	temp2->MoveUnit(reg->GetDummy());
+
+	if (Globals->LAIR_MONSTERS_EXIST || Globals->WANDERING_MONSTERS_EXIST) {
+		// Try to auto-declare all player factions unfriendly
+		// to Creatures, since all they do is attack you.
+		pFac->SetAttitude(monfaction, A_UNFRIENDLY);
+	}
+
+	return( 1 );
+}
+
+static void CreateQuest(ARegionList *regions, int monfaction)
+{
```

Comment:
I look forward to seeing your new ideas for quests :-)

Reply:
yes! i already have some. I like your visiting / killing code but instead of getting relics it gives you magic items and/or silver.